File: Atcoder_#337/ruby/b_extended_abc.rb
def main(input)
  s = input.strip
  # 正規表現を使って、拡張ABC文字列であるかどうかを判定します
  if s.match?(/\AA*B*C*\z/)
    puts 'Yes'
  else
    puts 'No'
  end
end

input = gets
main(input)

File: Atcoder_#337/javascript/b_extended_abc.js
function main(input) {
    const s = input.trim();
    // 正規表現を使って、拡張ABC文字列であるかどうかを判定します
    const isExtendedABCString = /^A*B*C*$/.test(s);

    if (isExtendedABCString) {
        console.log('Yes');
    } else {
        console.log('No');
    }
}

const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8');
main(input);
